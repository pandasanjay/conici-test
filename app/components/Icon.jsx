import React from "react";
import "./Icon.scss";
const Icon = ({ iconname }) => {
  return <span className={"sh-icon sh-i-" + iconname} />;
};
export default Icon;
