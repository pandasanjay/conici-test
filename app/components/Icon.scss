.sh-icon {
    width: 24px;
    height: 24px;
    display: inline-block;
    background-position: center;
    background-repeat: no-repeat;

    &.sh-i-location-uk {
        background-image: url("./../assets/location-uk.png");
    }

    &.sh-i-menu-bar {
        background-image: url("./../assets/menu-bars.png");
    }

    &.sh-i-icon-add-active {
        background-image: url("./../assets/icon-add-active.png");
    }

    &.sh-i-edit {
        background-image: url("./../assets/edit.png");
    }

    &.sh-i-edit-invert {
        background-image: url("./../assets/edit-invert.png");

    }

    &.sh-i-selected {
        background-image: url("./../assets/selected.png");

    }
}