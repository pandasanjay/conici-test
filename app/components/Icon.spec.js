import React from "react";
import Icon from "./Icon";
import { shallow } from "enzyme";
describe("Header component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Icon iconname="icon-name" />);
  });
  it("should have match with snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("should have class", function() {
    expect(component.find(".sh-icon.sh-i-icon-name").length).toEqual(1);
  });
});
