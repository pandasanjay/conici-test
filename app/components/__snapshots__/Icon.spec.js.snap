// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should have match with snapshot 1`] = `
<span
  className="sh-icon sh-i-icon-name"
/>
`;
